casper/src/main/scala/coop/rchain/casper/blocks/BlockProcessor.scala

```diff
@@ -19,7 +19,7 @@ import coop.rchain.shared.Log
   * Logic for processing incoming blocks
   * Blocks created by node itself are not held here, but in Proposer.
   */
-class BlockProcessor[F[_]: Concurrent](
+class BlockProcessor[F[_]: Concurrent: BlockDagStorage: CasperBufferStorage](
```

Comment:
Will this BlockProcessor be removed completely and replaced with BlockReceiver?
I'm curious why shardOfInterest validation is removed a bit lower?

Reply:
No, BlockProcessor will manage slashable validation with replay.

Shard will be validated in BlockReceiver so here it must be correct.